pkg/client/tx/client_test.go

```diff
@@ -569,6 +616,222 @@ func TestTxClient_SignAndBroadcast_MultipleMsgs(t *testing.T) {
 	t.SkipNow()
 }
 
+func TestTxClient_GasConfig(t *testing.T) {
+	var (
+		ctx             = context.Background()
+		blocksPublishCh = make(chan client.Block, 1)
+
+		// Standard test values
+		standardGasPrices = cosmostypes.NewDecCoins(
+			cosmostypes.NewDecCoin(volatile.DenomuPOKT, math.NewInt(1000)),
+		)
+		standardFeeAmount = cosmostypes.NewDecCoins(
+			cosmostypes.NewDecCoin(volatile.DenomuPOKT, math.NewInt(10000)),
+		)
+	)
+
+	tests := []struct {
+		name          string
+		options       []client.TxClientOption
+		expectError   bool
+		errorContains string
+		validateFee   func(t *testing.T, txBuilder cosmosclient.TxBuilder)
+	}{
+		{
+			name: "no gas params - should fail with error",
+			options: []client.TxClientOption{
+				tx.WithSigningKeyName(testSigningKeyName),
+			},
+			expectError:   true,
+			errorContains: "gas prices must be set",
+		},
+		{
+			name: "only gas prices provided - should use default gas settings",
+			options: []client.TxClientOption{
+				tx.WithSigningKeyName(testSigningKeyName),
+				tx.WithGasPrices(&standardGasPrices),
+			},
+			expectError: false,
+			validateFee: func(t *testing.T, txBuilder cosmosclient.TxBuilder) {
+				// Default gas * gas price = 200000 * 1000 = 200000000
+				feeCoins := txBuilder.GetTx().GetFee()
+				require.Equal(t, 1, len(feeCoins))
+				require.Equal(t, "200000000", feeCoins[0].Amount.String())
+				require.Equal(t, volatile.DenomuPOKT, feeCoins[0].Denom)
+			},
+		},
+		{
+			name: "fee amount provided - should override gas settings",
+			options: []client.TxClientOption{
+				tx.WithSigningKeyName(testSigningKeyName),
+				tx.WithFeeAmount(&standardFeeAmount),
```

Comment:
Just curious, would/should it be an error if you were to also provide a gas prices option here?